### DISCUSSION TITLE
Recycled stream showing zero flow in a system

### DISCUSSION THREAD

**Discussion by Uzair1shah** [BioSTEAMDevelopmentGroup/biosteam#168] [Q&A] [answered]
Hello,

I have been working on preprocessing area of a biorefinery and it requires recycle stream. The process flow diagram is as follows:
![Figure 2023-09-21 132011](https://github.com/BioSTEAMDevelopmentGroup/biosteam/assets/128337604/864fe523-669f-4577-8ffd-2a7985648d25)

The recycle stream, S113, shows some flow rate after the vibrating screen (UO108), but in mixer (UO106), the stream shows zero flow. The results are shown below:
Mixer, UO106:
![image](https://github.com/BioSTEAMDevelopmentGroup/biosteam/assets/128337604/d268df19-ef0e-4d6b-b06f-18f43278e478)
Vibrating screen, UO108 (where S113 has specific flowrate)
![image](https://github.com/BioSTEAMDevelopmentGroup/biosteam/assets/128337604/674efbb0-2118-44be-81bd-6fc87023ef2c)

The code for the recycled system is as follows:
```
S113 = Stream('S113', phase='s') #recycle stream
S114 = Stream('S114', phase='s')
UO106 = units.Mixer('UO106', ins=(S109, S113), outs=(S114))
UO106.simulate()
UO106.show(flow='kg/hr', composition = False, N=9)
r6 = UO106.results()
print(r6)


# Hammermill
S112 = Stream('S112', phase='s') #crushed dry feedstock
S115 = Stream('S115', phase='s') 

UO107 = units.HammerMill('UO107', ins=(S114), outs=(S115))

# Simulation of UO107 (Hammermill)
UO107.simulate()
UO107.show(flow='kg/hr', composition = False, N=9)
r7 = UO107.results()
print(r7)

#Vibrating Screen
S116 = Stream('S116', phase='s')
UO108 = VibratingScreen('UO108', ins=S115, outs=(S113,S116), split=0.1) 

# Simulation of UO107 (Vibrating Screen)
UO108.simulate()
UO108.show(flow='kg/hr', composition = False, N=9)
r8 = UO108.results()
print(r8)

Preprocessing = bst.main_flowsheet.create_system('Preprocessing')
Preprocessing.simulate()
Preprocessing.diagram('cluster', format='png')
```
So, I wanted to ask if there is any other way to solve the recycled stream? I looked into different examples in the documentation and tried different ways but I still get S113 to be zero in the mixer.  I am new to all of this and trying to learn BioSTEAM. Your help would be highly appreciated. Thanks a lot.

Best,
Uzair

**Comment by yoelcortes** [comment, marked as answer]
@Uzair1shah,

You are seeing a zero flow rate to the mixer because you are printing the streams before you simulate the system. The code you sent me doesn't actually run, but I edited it anyway to convey that you should print results after simulating the systems (which solves recycle loops):

```python
S113 = Stream('S113', phase='s') #recycle stream
S114 = Stream('S114', phase='s')
UO106 = units.Mixer('UO106', ins=(S109, S113), outs=(S114))

# Hammermill
S112 = Stream('S112', phase='s') #crushed dry feedstock
S115 = Stream('S115', phase='s') 

UO107 = units.HammerMill('UO107', ins=(S114), outs=(S115))

#Vibrating Screen
S116 = Stream('S116', phase='s')
UO108 = VibratingScreen('UO108', ins=S115, outs=(S113,S116), split=0.1) 

Preprocessing = bst.main_flowsheet.create_system('Preprocessing')
Preprocessing.simulate()
Preprocessing.diagram('cluster', format='png')

UO106.show(flow='kg/hr', composition = False, N=9)
r6 = UO106.results()
print(r6)

UO107.show(flow='kg/hr', composition = False, N=9)
r7 = UO107.results()
print(r7)

UO108.show(flow='kg/hr', composition = False, N=9)
r8 = UO108.results()
print(r8)
```

Thanks!

**Comment by Uzair1shah** [comment]
Thanks a lot. That worked and it also makes sense now. I was simulating the individual unit operations. Thank you so much again.
